test/test_torch.py

```diff
@@ -11001,6 +11001,69 @@ def test_exponential(self, device, dtype):
         with self.assertRaises(RuntimeError):
             torch.empty((1,), device=device, dtype=dtype).exponential_(-0.5)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @dtypes(*(torch.testing.get_all_fp_dtypes(include_half=False) +
+              torch.testing.get_all_complex_dtypes()))
+    @dtypesIfCUDA(*(torch.testing.get_all_fp_dtypes(include_half=True) +
+                    torch.testing.get_all_complex_dtypes()))
+    def test_exp(self, device, dtype):
+        for v in (2, -2) + ((1j, 1 + 1j) if dtype.is_complex else ()):
+            if dtype == torch.bfloat16:
+                # Currently multiply a bfloat16 type with floating-point causes error:
+                #   RuntimeError: dtype != ScalarType::Undefined INTERNAL ASSERT FAILED at
+                #   "/pytorch/aten/src/ATen/native/TensorIterator.cpp":125, please report a bug to PyTorch.
+                # We skip bfloat16 for now, but we should fix it.
+                if not device.startswith('xla'):
+                    with self.assertRaises(RuntimeError):
+                        torch.tensor(v, dtype=dtype, device=device) * torch.arange(18, device=device)
+                    return
+
+            a = torch.tensor(v, dtype=dtype, device=device) * torch.arange(18, device=device) / 3 * math.pi
+            expected = np.exp(a.cpu().numpy())
+            actual = torch.exp(a)
+            self.assertEqual(actual, torch.from_numpy(expected))
+
```

Comment:
```
if device.startswith('cpu'):
   return
```
since the remaining code is only run for complex dtypes and cpu calculations are buggy

Reply:
The reason we explicitly add tests incorrect results is that, when we actually start correcting the issues, we would be aware that we need to add tests for them, and know where and what to add. I learned this long time ago: https://github.com/pytorch/pytorch/pull/27248#discussion_r335127837

Reply:
`if self.device_type == 'cpu':` is preferred, although it's not as discoverable.